x-pack/plugins/observability_solution/investigate_app/public/hooks/use_fetch_investigation.ts

```diff
@@ -16,7 +16,7 @@ import { investigationKeys } from './query_key_factory';
 import { useKibana } from './use_kibana';
 
 export interface Params {
-  id: string;
+  id?: string;
```

Comment:
In which scenario `id` would be undefined? also, this is not part of the PR, but do you know why do we need `initialInvestigation` in params?

Reply:
id would be undefined when we open the [form in "creation" mode](https://github.com/elastic/kibana/pull/192208/files/3d5e6b499b04652e44186b557ca25ac68eef328b#diff-6078e390657ff0e1aea47cb7590f063a1a555d7a74aa6da931ebfc758a6e2314R52-R59), i.e. when the investigationId is undefined. 

As for the `initialInvestigation`, we are leverage the `initialData` feature from [React Query](https://tanstack.com/query/v4/docs/framework/react/guides/initial-query-data#using-initialdata-to-prepopulate-a-query) to avoid showing the loading state when we already have the investigation from a parent component.

And we use that in the main investigation details page component, when we fetch the investigation (using the hook) and either show some loading/error state or populate the Investigation Context Provider with it.
This Context Provider is actually reusing the same hook (in order to handle refetch on updates), and to avoid handling the loading/error state again, we pre-populate the data with what we just fetched.

Reply:
thanks for the explanation